backend/corpora/common/utils/corpora_constants.py

```diff
@@ -7,31 +38,48 @@ class CorporaConstants(object):
     # Constants related to the metadata requirements of submitting a dataset to Corpora. Note that these metadata
     # fields are CASE-SENSITIVE.
     REQUIRED_OBSERVATION_METADATA_FIELDS = [
-        "tissue",
-        "assay",
-        "disease",
-        "cell_type",
-        "sex",
-        "ethnicity",
-        "developmental_stage",
+        TypedMetadata(field_name="tissue", required_type=str),
+        TypedMetadata(field_name="assay", required_type=str),
+        TypedMetadata(field_name="disease", required_type=str),
+        TypedMetadata(field_name="cell_type", required_type=str),
+        TypedMetadata(field_name="sex", required_type=SEX_VALID_ENUM_VALUES),
+        TypedMetadata(field_name="ethnicity", required_type=str),
+        TypedMetadata(field_name="developmental_stage", required_type=str),
     ]
     REQUIRED_OBSERVATION_ONTOLOGY_METADATA_FIELDS = [
-        "tissue_ontology",
-        "assay_ontology",
-        "disease_ontology",
-        "cell_type_ontology",
-        "ethnicity_ontology",
-        "developmental_stage_ontology",
+        TypedMetadata(field_name="tissue_ontology", required_type=UBERON_ONTOLOGY),
+        TypedMetadata(field_name="assay_ontology", required_type=EFO_ONTOLOGY),
+        TypedMetadata(field_name="disease_ontology", required_type=MONDO_ONTOLOGY, valid_alternative="PATO:0000461"),
+        TypedMetadata(field_name="cell_type_ontology", required_type=CELL_ONTOLOGY, valid_alternative=""),
+        TypedMetadata(field_name="ethnicity_ontology", required_type=HANCESTRO_ONTOLOGY, valid_alternative="na"),
+        TypedMetadata(
+            field_name="developmental_stage_ontology", required_type=HSAPDV_ONTOLOGY, valid_alternative="EFO:0000399"
+        ),
+    ]
+    REQUIRED_DATASET_METADATA_FIELDS = [
+        TypedMetadata(field_name="organism", required_type=str),
+        TypedMetadata(field_name="organism_ontology", required_type=NCBI_TAXON_ONTOLOGY),
+        TypedMetadata(field_name=LAYER_DESCRIPTIONS, required_type=dict),
+    ]
+    REQUIRED_DATASET_PRESENTATION_METADATA_FIELDS = [
+        TypedMetadata(field_name="title", required_type=str),
+        TypedMetadata(field_name="contributors", required_type=list),
+        TypedMetadata(field_name="preprint_doi", required_type=str),
+        TypedMetadata(field_name="publication_doi", required_type=str),
+    ]
+
+    REQUIRED_DATASET_PRESENTATION_HINTS_METADATA_FIELDS = [
+        TypedMetadata(field_name="color_map", required_type=list),
```

Comment:
I think this is supposed to be a dict, a dict with lists as values

Reply:
Done